src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/models/sweeper/SweptManagedIndex.kt

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.indexstatemanagement.models.sweeper
+
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.models.ChangePolicy
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.models.ManagedIndex
+import org.elasticsearch.common.bytes.BytesReference
+import org.elasticsearch.common.xcontent.LoggingDeprecationHandler
+import org.elasticsearch.common.xcontent.NamedXContentRegistry
+import org.elasticsearch.common.xcontent.XContentHelper
+import org.elasticsearch.common.xcontent.XContentParser
+import org.elasticsearch.common.xcontent.XContentParser.Token
+import org.elasticsearch.common.xcontent.XContentParserUtils.ensureExpectedToken
+import org.elasticsearch.common.xcontent.XContentType
+import java.io.IOException
+
+/**
+ * Data class to hold partial [ManagedIndex] data.
+ *
+ * This data class is used in the [com.amazon.opendistroforelasticsearch.indexstatemanagement.ManagedIndexSweeper]
+ * to hold partial data when reading in the [ManagedIndex] document from the index.
+ */
+data class SweptManagedIndex(
```

Comment:
The name `ManagedIndex` is kind of confusing since it's not an Index.  It's more of a Job that manages an index... Can we come up with something else?

Reply:
Renaming to ManagedIndexConfig